ecommerce-app/src/pages/index.js

```diff
@@ -0,0 +1,53 @@
+// pages/index.js
+import React, { useEffect, useState } from "react";
+import Link from "next/link";
+import styles from "../styles/index.module.scss";
+
+const Home = () => {
+  const [categories, setCategories] = useState([]);
+  const [products, setProducts] = useState([]);
+
+  useEffect(() => {
+    // Fetch categories
+    fetch("/api/categories")
```

Comment:
Prefer using async-await. Create a separate hook that handles fetch related stuff